contracts/sol6/Dao/mock/KyberFeeHandlerWrapper.sol

```diff
@@ -87,18 +76,16 @@ contract KyberFeeHandlerWrapper is DaoOperator {
             KyberFeeHandlerData[] memory kyberFeeHandlerArray = kyberFeeHandlersPerToken[supportedTokens[i]];
             uint256 endKyberFeeHandlerId = (endKyberFeeHandlerIndex >= kyberFeeHandlerArray.length) ?
                 kyberFeeHandlerArray.length - 1: endKyberFeeHandlerIndex;
+            amounts = new uint256[](endKyberFeeHandlerId - startKyberFeeHandlerIndex + 1);
+
             // iteration starts from endIndex, differs from claiming reserve rebates and platform wallets
             for (uint256 j = endKyberFeeHandlerId; j >= startKyberFeeHandlerIndex; j--) {
                 KyberFeeHandlerData memory kyberFeeHandlerData = kyberFeeHandlerArray[j];
                 if (j == 0 || kyberFeeHandlerData.startEpoch < epoch) {
-                    amountWei = amountWei.add(
-                        kyberFeeHandlerData.kyberFeeHandler.claimStakerReward(staker, epoch)
-                    );
+                    amounts[j] = kyberFeeHandlerData.kyberFeeHandler.claimStakerReward(staker, epoch);
                     break;
                 } else if (kyberFeeHandlerData.startEpoch == epoch) {
-                    amountWei = amountWei.add(
-                        kyberFeeHandlerData.kyberFeeHandler.claimStakerReward(staker, epoch)
-                    );
+                    amounts[j] = kyberFeeHandlerData.kyberFeeHandler.claimStakerReward(staker, epoch);
```

Comment:
shouldn't we claim for another epoch here?

Reply:
Make another call for the previous epoch. 1 epoch at a time =) Unless u mean claim from another feeHandler?

Reply:
yes
I mean another fee handler.

lets assume epoch was 25.

some fees were sent to fee handler 4 in the list. and then fee handler 5 was added.

so fee handler 5 startEpoch == epoch. (25)
and almost sure that feeHandler 4 has some fees as well.

since most chances the new fee handler wasn't added exactly at epoch start.

Reply:
Yes, that is correct, will continue with the loop and claim from 1 (or more) previous feeHandlers. The "overlapping epochs" test covers this scenario. 

Let's say feeHandler5 has a bug, then feeHandler6 was added in the same epoch (25). So when u claim from epoch 25, have to claim from feeHandlers 4, 5 and 6.

Reply:
OK
so if startEpoch == epoch we don't break and continue to claim

makes sense